launch/launch/output_handler.py

```diff
@@ -0,0 +1,196 @@
+import sys
+
+
+class OutputHandler(object):
+
+    def __init__(self):
+        self.print_mutex = None
+        self.line_prefix = None
+
+    def get_description(self):
+        raise NotImplemented
+
+    def set_print_mutex(self, print_mutex):
+        self.print_mutex = print_mutex
+
+    def set_line_prefix(self, line_prefix):
+        self.line_prefix = line_prefix
+
+    def process_init(self):
+        pass
+
+    def support_stderr2stdout(self):
+        return False
+
+    def on_message_received(self, lines):
+        self.on_stdout_received(lines)
+
+    def on_stdout_received(self, data):
+        pass
+
+    def on_stderr_received(self, data):
+        pass
+
+    def flush(self):
+        pass
+
+    def process_cleanup(self):
+        pass
+
+
+class CompositeOutputHandler(object):
+
+    def __init__(self, output_handlers):
+        self.output_handlers = output_handlers
+
+    def get_description(self):
+        return ', '.join([h.get_description() for h in self.output_handlers])
+
+    def support_stderr2stdout(self):
+        return all([h.support_stderr2stdout() for h in self.output_handlers])
+
+    def __getattr__(self, name):
+        def wrapper(*args, **kwargs):
+            [getattr(h, name)(*args, **kwargs) for h in self.output_handlers]
```

Comment:
These are all callback functions. None of them currently returns anything. I would rather not return anything then a list of `None`s.